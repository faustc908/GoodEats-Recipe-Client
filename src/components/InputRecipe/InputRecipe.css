.input {
  text-align: center;
  padding: 20px;
  border-bottom: 2px solid black;
  margin-top: 40px;
  margin-bottom: 40px;
}

.inputButton {
  background: rgb(45, 48, 245);
  border: none;
  padding: 5px 40px;
  color: white;
  border: 2px solid rgba(17, 3, 82, 0.601);
  border-radius: 20px;
  margin-left: 10px;
}

.inputButton:hover {
  background-color: rgba(134, 133, 133, 0.601);
  color: white;
}

.appDesc {
  font-style: italic;
}
