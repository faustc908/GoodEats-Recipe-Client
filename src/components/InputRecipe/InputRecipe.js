import React, { useState } from "react";
import "./InputRecipe.css";

const InputRecipe = ({ updateRecipe }) => {
  const [description, setDescription] = useState("");

  // Post function to post recipes

  const onSubmitForm = async (e) => {
    e.preventDefault();
    try {
      const body = { description };

      const header = {
        method: "POST",
        headers: { "Content-Type": "application/json" },
        body: JSON.stringify(body),
      };
      alert("saving");
      await fetch("http://localhost:8000/recipe", header).then(async function (
        response
      ) {
        const data = await response.json();
        alert(data.description);
        updateRecipe();
      });
    } catch (err) {
      console.error(err.message);
    }
  };

  return (
    <div className="input">
      <h1 className="appDesc">User Recipes</h1>
      <p>Enter your own favorite recipes below!</p>
      <form onSubmit={onSubmitForm}>
        <input
          className="inputBox"
          type="text"
          value={description}
          onChange={(e) => setDescription(e.target.value)}
        />
        <button className="inputButton" type="button" onClick={onSubmitForm}>
          Add
        </button>
      </form>
    </div>
  );
};

export default InputRecipe;
