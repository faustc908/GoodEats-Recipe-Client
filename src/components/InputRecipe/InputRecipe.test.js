import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import InputRecipe from './InputRecipe'
import ReactDOM from 'react-dom';

describe(`InputRecipe component`, () => {

    it('renders a InputRecipe by default', () => {
        const wrapper = shallow(<InputRecipe />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<InputRecipe />, div);
        ReactDOM.unmountComponentAtNode(div);
    });
})