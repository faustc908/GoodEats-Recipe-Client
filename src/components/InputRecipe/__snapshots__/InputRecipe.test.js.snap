// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputRecipe component renders a InputRecipe by default 1`] = `
<div
  className="input"
>
  <h1
    className="appDesc"
  >
    User Recipes
  </h1>
  <p>
    Enter your own favorite recipes below!
  </p>
  <form
    onSubmit={[Function]}
  >
    <input
      className="inputBox"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="inputButton"
      onClick={[Function]}
      type="button"
    >
      Add recipe
    </button>
  </form>
</div>
`;
